pkg/client/client.go

```diff
@@ -2055,3 +2023,24 @@ func deleteOptions(dp metav1.DeletionPropagation) metav1.DeleteOptions {
 		PropagationPolicy: &dp,
 	}
 }
+
+// pollUntil is a wrapper around wait.PollUntilContextTimeout that allows adding the passed lastError into
+// the final error if if set by the condition, this would add more context to the "context deadline exceeded" error e.g..
+func pollUntil(ctx context.Context, timeout, interval time.Duration, lastError *error, condition wait.ConditionWithContextFunc) error {
+	if err := wait.PollUntilContextTimeout(ctx, interval, timeout, false, condition); err != nil {
+		if lastError != nil && *lastError != nil {
+			err = fmt.Errorf("%w: lastError: %w", err, *lastError)
+		}
+		return err
+	}
+	return nil
+}
+
+func Poll(ctx context.Context, lastError *error, condition wait.ConditionWithContextFunc) error {
+	return pollUntil(ctx, 5*time.Minute, 3*time.Second, lastError, condition)
```

Comment:
This sets the same interval/timeout to all polls (which is not the case for interval currently, but it's better to have the same sane values set for all of them)

Reply:
I'd prefer to keep the same values as before to avoid unintended changes (especially regarding timeouts).

One solution could be to extend the signature with functional options:

```go
type pollOptions struct {
  timeout time.Duration
  interval time.Duration
  lastError *error
}

func withPollTimeout(d time.Duration) func(o *pollOptions) {
    return func(o *pollOptions) {
        o.timeout = d
    }
}

func withPollInterval(d time.Duration) func(o *pollOptions) {
    return func(o *pollOptions) {
        o.interval = d
    }
}

func withLastError(e *error) func(o *pollOptions) {
    return func(o *pollOptions) {
        o.lastError = e
    }
}

func Poll(ctx context.Context, condition wait.ConditionWithContextFunc, options ...pollOptions) error {
    opts := pollOptions{timeout: 5 * time.Minute, interval: time.Second}
    for _, o := range options {
       o(&opts)
    }
...
}

```

Reply:
Actually, the timeouts weren't changed (5mins for "deploy" requests and 10 for delete)
I just aligned the intervals to 3s (was aiming for 5s as well) for deploy and 5s for delete to simplify things without affecting the api server.
But if you think we should keep intervals as is, we can allow customizing.

Reply:
yes I'd prefer to keep things as-is for now. We can still look at refactoring afterwards.

Reply:
okk, if you could take a look now.